test/functional.js

```diff
@@ -1326,6 +1326,227 @@ module.exports = {
+                            'runtime.js',
+                            'main.js',
+                            'manifest.json'
+                        ]);
+
+                    expect(path.join(config.outputPath, 'assets')).to.be.a.directory()
+                        .with.files([
+                            'symfony_logo.png',
+                            'symfony_logo_alt.png',
+                            'Roboto.woff2',
+                        ]);
+
+                    webpackAssert.assertManifestPath(
+                        'build/main.js',
+                        '/build/main.js'
+                    );
+
+                    webpackAssert.assertManifestPath(
+                        'build/assets/symfony_logo.png',
+                        '/build/assets/symfony_logo.png'
+                    );
+
+                    webpackAssert.assertManifestPath(
+                        'build/assets/symfony_logo_alt.png',
+                        '/build/assets/symfony_logo_alt.png'
+                    );
+
+                    webpackAssert.assertManifestPath(
+                        'build/assets/Roboto.woff2',
+                        '/build/assets/Roboto.woff2'
+                    );
+
+                    done();
+                });
+            });
+
+            it('Copy folder and subdirectories with versioning enabled to the specified location', (done) => {
+                const config = createWebpackConfig('www/build', 'production');
+                config.addEntry('main', './js/no_require');
+                config.setPublicPath('/build');
+                config.copyFiles({
+                    from: './images',
+                    to: 'images/[path][name].[hash:8].[ext]',
+                    includeSubdirectories: true
+                });
+
+                testSetup.runWebpack(config, (webpackAssert) => {
+                    expect(config.outputPath).to.be.a.directory()
+                        .with.files([
+                            'entrypoints.json',
+                            'runtime.js',
+                            'main.js',
+                            'manifest.json',
+                        ]);
+
+                    expect(path.join(config.outputPath, 'images')).to.be.a.directory()
+                        .with.files([
+                            'symfony_logo.ea1ca6f7.png',
+                            'symfony_logo_alt.f27119c2.png',
+                        ]);
+
+                    expect(path.join(config.outputPath, 'images', 'same_filename')).to.be.a.directory()
+                        .with.files([
+                            'symfony_logo.f27119c2.png',
+                        ]);
+
+                    webpackAssert.assertManifestPath(
+                        'build/main.js',
+                        '/build/main.js'
+                    );
+
+                    webpackAssert.assertManifestPath(
+                        'build/images/symfony_logo.png',
```

Comment:
Ok, so to use this stuff, people would:

```
<img src="{{ path('build/images/symfony_logo.png') }}">
```

Right? They would basically "know" that the copy function puts things into the `build/` directory, so then they reference that path. Gosh, and the `manifest.json` is even smart enough to create the key *without* the `[hash]` part of the filename? That's incredible!

Reply:
Yep, it's not different than doing a `require(...)` inside of a real entry after all :)

It doesn't really remove the `[hash]` part but simply uses the original name of the file which is something that wasn't possible to do with the `CopyWebpackPlugin` since it didn't expose it.